src/main/kotlin/com/exactpro/th2/kafka/client/RawMessageProcessor.kt

```diff
@@ -24,30 +27,38 @@ class RawMessageProcessor(
 ) : AutoCloseable {
     private val messageQueue: BlockingQueue<RawMessage.Builder> = LinkedBlockingQueue()
     private val batchQueue: BlockingQueue<RawMessageBatch> = LinkedBlockingQueue()
+    private val batchFlusherExecutor = Executors.newSingleThreadScheduledExecutor()
 
     private val messageReceiverThread = thread(name = "message-receiver") {
-        val batchFlusherExecutor = Executors.newSingleThreadScheduledExecutor()
-        val batchBuilder = RawMessageBatch.newBuilder()
-        var flusherFuture: Future<*> = CompletableFuture.completedFuture(null)
-        val lock = ReentrantLock()
-
-        fun enqueueBatch() = lock.withLock {
-            if (batchBuilder.messagesCount > 0) {
-                flusherFuture.cancel(false)
-                batchQueue.add(batchBuilder.build())
-                batchBuilder.clear()
-            }
-        }
+        val builders: ConcurrentMap<String, BatchHolder> = ConcurrentHashMap()
```

Comment:
Why do you need a concurrent map here? It is not shared between threads as I can see